src/GitHub.App/Models/RepositoryModel.cs

```diff
@@ -1,28 +1,68 @@
 ﻿using GitHub.Primitives;
-using GitHub.UI;
+using NullGuard;
+using System;
+using System.Globalization;
 
 namespace GitHub.Models
 {
-    public class RepositoryModel : IRepositoryModel
+    public class RepositoryModel : SimpleRepositoryModel, IRepositoryModel, IEquatable<RepositoryModel>
     {
-        public RepositoryModel(string name, string cloneUrl, bool isPrivate, bool isFork,  IAccount ownerAccount)
+        public RepositoryModel(string name, UriString cloneUrl, bool isPrivate, bool isFork,  IAccount ownerAccount)
+            : base(name, cloneUrl)
         {
-            Name = name;
             Owner = ownerAccount;
-            CloneUrl = cloneUrl;
-            Icon = isPrivate
-                ? Octicon.@lock
-                : isFork
-                    ? Octicon.repo_forked
-                    : Octicon.repo;
+            SetIcon(isPrivate, isFork);
         }
 
-        public string Name { get; private set; }
-
         public IAccount Owner { get; private set; }
+        public override int GetHashCode()
+        {
+            return (Owner != null ? 0 : Owner.GetHashCode()) ^ base.GetHashCode();
```

Comment:
I think you meant `(Owner == null ? 0 : Owner.GetHashCode()) ^ base.GetHashCode();` on this line. Otherwise you could get a `NullReferenceException`